name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/InvestmentHeatmapWidget.java

```diff
@@ -69,8 +69,12 @@ public EarningsConfig(WidgetDelegate<?> delegate)
 
     enum GrossNetType
     {
-        NET(Messages.LabelNet, t -> t.getMonetaryAmount()), //
-        GROSS(Messages.LabelGross, t -> t.getGrossValue());
+        /***
+         * Net investment = gross investment minus costs
+         * Gross investment = total of all investments made
+         */
+        NET(Messages.LabelNetInvest, t -> t.getGrossValue()),
+        GROSS(Messages.LabelGrossInvest, t -> t.getMonetaryAmount());
```

Comment:
Hier scheinen wir unterschiedliches Verständnis zu haben. Das sollten wir diskutieren.

Du schreibst **Net investment = gross investment minus costs**.
Das ist aber für mich genau der ```getMonetaryAmount```.
getMonetaryAmount liefert die Investment (sagen wir 1000 EUR) minus Gebühren und Steuern (10 EUR) und damit 990 EUR.

Die Method ```getGrossValue``` liefert den Wert *ohne* Gebühren und Steuern. Also im Fall eines Kaufes eben 1000 EUR. 

Oder wo verstehe ich das falsch?

Reply:
Hallo @buchen,
schauen wir uns das nochmal an.

Ich kaufe eine Aktie X für 100€. Die Gebühr etc. liegt bei 5€.

Net investment = gross investment minus costs
100 € = 105€ - 5€

Gross investment = total of all investments made
105€ 

Ergebnis des Investment-Widget:
![grafik](https://user-images.githubusercontent.com/45203494/119258179-e97bf380-bbc8-11eb-8688-d340facde42e.png)

Wie du sehen kannst, sind die Ergebnisse des aktuellen Widget genau verkehrt herum, oder ich bin auf dem Holzweg.

Reply:
Nein, Du Bist nicht auf dem Holzweg, sondern wohl eher ich.

Ich glaube ich verwende den ```GROSS_VALUE``` in der Buchung einfach falsch. Die Method ```PortfolioTransaction#getGrossValue``` liefert den Betrag ohne Steuern und Gebühren.

Das heißt, ich muss eigentlich noch mal alle Brutto/Netto Verwendungen prüfen 😟 

[Hier heißt es](https://www.rechnungswesen-verstehen.de/lexikon/netto-und-brutto.php):
> Netto bedeutet rein, also ist es der berinigte Teil und das bedeutet ohne Mehrwert- oder Umsatzsteuer. Brutto ist das Gesamte und enthält daher die Mehrwert- und Umsatzsteuer. Somit ist Brutto immer höher als Netto. Grundsätzlich gilt: Netto ist ein Teil des Bruttos.

Demnach müsste die Buchungskomponente wohl eher NET_VALUE heißen. Und die Methoden #getNetValue.

Beim Kauf: Brutto: 105, Netto: 100
Beim Verkauf: Brutton: 95, Netto: 100

Reply:
Wenn ich [hier schaue](http://www.wirtschaftslexikon24.com/d/bruttodividende/bruttodividende.htm), ist es aber bei Dividenden genau andersrum:

```
Brutto-Dividende                              100
Einkommensteuer  (z.B. Steuersatz 30%)         30
Dividende nach Steuer = verfügbare Dividende   70
(aus der Sicht der Anteils­eigner gelegentlich auch als
"Nettodividende" bezeich­net)
```
Die Nettodividende ist das was übrig bleibt...

Reply:
Und auch beim [Verkauf](https://teamaftermarket.com/1390-net-proceeds) würde mal wohl sagen:

> Der Nettoerlös ist der endgültige Geldbetrag, den ein Verkäufer in Bezug auf die Veräußerung eines Vermögenswerts abzüglich aller damit verbundenen Kosten wie Provisionen, Gebühren usw. erhalten kann. Er wird berechnet, indem alle Verkaufskosten von abgezogen werden der Verkaufspreis eines Vermögenswerts.

Reply:
Vielleicht benennen wir Brutto/Netto einfach in "ohne Gebühre/Steuern" und "unter Berücksichtigung von Gebühren und Steuern" um.

Denn Brutto/Netto ist eine Frage der Betrachtung:

Beim Investment ist das *Nettoinvestment* der Betrag ohne Kosten, also die 100 Euro.
Beim Erlös ist der *Nettoerlös* der Betrag abzüglich der Steuern und Gebühren, also die 95 Euro.

Reply:
> Vielleicht benennen wir Brutto/Netto einfach in "ohne Gebühre/Steuern" und "unter Berücksichtigung von Gebühren und Steuern" um.

Ich finde, das ist genauso mehrdeutig …

Es geht besser bei Transaktionen, bei denen die Gebühren „in die gleiche Richtung“ wirken (Kauf für 100 Euro, es werden zusätzlich 5 Euro Gebühren in Rechnung gestellt), als bei solchen, bei denen sie „in die andere Richtung“ wirken (Verkauf für 100 Euro, es werden aber wegen Gebühren 5 Euro weniger gutgeschrieben, also nur 95 Euro).

Reply:
Hallo @buchen 
ja, das ist in der Tat kniffelig. 🤔 
Bei dem Investment-Widget hatte ich deswegen, die Übersetzung erweitert in `Bruttoinvestition `und `Nettoinvestition`.

Somit sollte keine Verwechslung zwischen der normalen kaufmännischen Bezeichnung `Brutto `und `Netto `kommen.

Eine zusätzliche klare Variablenumgebung, die zu Berechnung führt, wäre natürlich sehr elegant in source. 
`NET_INVEST `und `BRUTTO_INVEST` mit der entsprechenden Formel/Berechnung, rundet das ganze noch ab.

Beim `Kauf `und `Verkauf `würde `Brutto `und `Netto `bestehen bleiben.

Bei Dividenden wäre `NET_DIVIDEND `und `BRUTTO_DIVIDEND` mit den Übersetzungen `Brutto-Dividende` und `Netto-Dividende` das äquivalent zum Investment-Widget.

Wäre das ein Lösungsansatz?

Reply:
> Es geht besser bei Transaktionen, bei denen die Gebühren „in die gleiche Richtung“ wirken (Kauf für 100 Euro, es werden zusätzlich 5 Euro Gebühren in Rechnung gestellt), als bei solchen, bei denen sie „in die andere Richtung“ wirken (Verkauf für 100 Euro, es werden aber wegen Gebühren 5 Euro weniger gutgeschrieben, also nur 95 Euro).

Das ist auch ein Punkt der mir hier Bauchschmerzen bereitet:

In dem Widget kann man sich entweder die **Investitionen** (BUY, DELIVERY_INBOUND) oder die **Investitionen, minus Verkaufserlöse** (also BUY, SELL, inbound & outbound DELIVERY) anzeigen lassen.

Ich finde es dann komisch, wenn beim Kauf die Steuern nicht betrachtet werden (Nettoinvestitionen - also ```PortfolioTransaction#getGrossValue``` und beim Verkauf die Steuern aber einbezogen werden (Nettoerlöse - also ```PortfolioTransaction#getMonetaryAmount```. Das fühlt sich für mich nicht konsistent an. Darum auch mein Vorschlag zu "vor Steuern/Gebühren" und "nach Steuern/Gebühren" anstelle von "Brutto/Netto".

> Eine zusätzliche klare Variablenumgebung, die zu Berechnung führt, wäre natürlich sehr elegant in source.

Ich glaube im Source will ich das jetzt doch nicht umbenennen. ```Transaction.Unit.Type.GROSS_VALUE``` muss man dann immer interpretieren als "vor Steuern und Gebühren". Beim Kauf ist es das Nettoinvestment, beim Verkauf ist es der Bruttoerlös, bei der Dividende die Bruttodividende. Schön wäre natürlich wenn es einen neutralen Begriff geben würde, aber direkt fällt mir nix ein. PRE_TAX_AND_FEES liest sich umständlich. NET_VALUE wäre vielleicht schöner im Sinne der Definition (der "reine" Wert - siehe oben), aber macht die Konfusion nicht besser.

Reply:
Ich denke ich werde jetzt erst mal nur das Label umbenennen - das macht es etwas besser, wenn auch noch nicht so richtig gut. Vor allem würde ich gerne heute mal wieder eine Release veröffentlichen... da muss ich mich ranhalten.

Reply:
Das:

```
        NET("nach Steuern/Gebühren (Bruttoinvestionen/Nettoerlöse)", t -> t.getMonetaryAmount()), //
        GROSS("vor Steuern/Gebühren (Nettoinvestitionen/Bruttoerlöse)", t -> t.getGrossValue());
```

finde ich dann auch zu viel. Aber man sieht, dass Brutto/Netto eben in die andere Richtung wirkt.